client/index.html

```diff
@@ -1,119 +1,172 @@
-  		<canvas id="ship" width="600" height="360">
-
-    </main>
-
-   <!-- <script src='/client/js/main.js'></script> -->
-   <script>
     document.addEventListener("DOMContentLoaded", function(event) {
-      let x = 200;
-      let y = 200;
-      let width = 57;
-      let height = 101;
-
-      let shipCanvas = document.getElementById('ship');
-      let shipContext = shipCanvas.getContext('2d');
-      let imgSpaceship = new Image();
-
-      let map = [ false, false, false, false ];
-
-      var clearScreen = function () {
-        shipContext.clearRect(0, 0, 600, 360 );
+      let canvas = document.getElementById("ship");
+      let context = canvas.getContext("2d");
+
+      let canvasWidth = canvas.width;
+      let canvasHeight = canvas.height;
+
+      //  player only
+      let rotation = 0;
+      let x = 50;
+      let y = 50;
+      // let facingX = 0;
+      // let facingY = 0;
+      // let movingX = 0;
+      // let movingY = 0;
+      let width = 30;  // drawn element width
+      let height = 30; // drawn element height
+      let W = width/2;
+      let H = height/2;
+      // let rotationalVelocity = 5; //how many degrees to turn the ship
+      // let thrust = .06;
+      let map = [];
+      let shipState = 0; //0 = static, 1 = thrust
+
+      let drawCanvas = (x, y, rotation) => {
+        // background
+        // context.fillStyle = '#000000';
+        // context.fillRect(0, 0, canvasWidth, canvasHeight);
+        // context.fillStyle = '#ffffff';
+        //  transform
+        let radians = convertToRadians(rotation);
+        context.save();    // save the current stack
+        context.setTransform(1,0,0,1,0,0); //  reset to identity
+        //  canvas origin Player center
+
+        context.translate(x+W,y+H);
+        context.rotate(radians);
+        //drawShip
+        context.strokeStyle = '#fff';
+        context.beginPath();
+        // draw the ship
+        // set rotation to image center
+        context.moveTo(-15,-15);
+        context.lineTo(15,0);
+        context.moveTo(15,1);
+        context.lineTo(-15,15);
+        context.lineTo(1,1);
+        context.moveTo(1,-1);
+        context.lineTo(-15,-15);
+        context.lineWidth=2;
+        context.stroke();
+        context.closePath();
+        //  restore context
+        context.restore();
+        //update the shipState
+        shipState++;
+        if (shipState==1) {
+          //draw thrust
+          context.moveTo(8,13);
+          context.lineTo(11,13);
+          context.moveTo(9,14);
+          context.lineTo(9,18);
+          context.moveTo(10,14);
+          context.lineTo(10,18);
+        }
+        if (shipState >1) {
+           shipState=0;
+        }
       }
 
-      //  define Ship class
-      let Ship = function () {
-        this.x = 50;
-        this.y = 50;
-        this.h = 50;
-        this.w = 50;
-        this.speed = 0;
-      }
+      var socket = io();
+      socket.on('newPosition', function (data) {
```

Comment:
Just commenting here: we're evaluating the movement on the client, and then notifying the server about the new position. This means that I can hack my instance client so I can cheat :D
I don't want you to change it, but I want you to have it in mind.